packages/validator/src/services/validatorStore.ts

```diff
@@ -182,7 +182,7 @@ export class ValidatorStore {
     // Duties are filtered before-hard by doppelganger-safe, this assert should never throw
     this.assertDoppelgangerSafe(pubkey);
 
-    const proposerDomain = this.config.getDomain(DOMAIN_BEACON_PROPOSER, blindedOrFull.slot);
+    const proposerDomain = this.config.getDomain(currentSlot, DOMAIN_BEACON_PROPOSER, blindedOrFull.slot);
```

Comment:
Should this really be the currentSlot? Without checking the specs, sounds it should be the state slot, so block's slot

Reply:
currentslot should be same as blindedorfull slot same as state slot for signing block here

Reply:
_should_ is not enough, this can lead to bad signatures on the fork slot if there are clock issues. There's no solid reason to not use `blindedOrFull.slot`

Reply:
updated :+1: